### PR TITLE
Add experiments focused on dogstatsd's string interner

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

This commit replaces the _nodist variants of the `uds_dogstatsd_to_api` experiment with a set of parameterized string interner experiments, varying both total contexts and throughput along two scales. The goal here is to understand how, at 8MiB/sec throughput, the variation of contexts impacts the stringInterner hotspot and along throughput to achieve coverage on a small range of contexts.

REF SMPTNG-12
REF #19852
REF #19882
REF #19990

### CODE REVIEW

**Review by scottopell — CHANGES_REQUESTED** [review]

File: test/regression/cases/dogstatsd_string_interner_128MiB_100/experiment.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+optimization_goal: ingress_throughput
+erratic: false
+
+environment:
+  DD_TELEMETRY_ENABLED: true
+
+profiling_environment:
+  DD_INTERNAL_PROFILING_BLOCK_PROFILE_RATE: 10000
```
[the remark this answers is not in the document]
Comment (scottopell) [reply test/regression/cases/dogstatsd_string_interner_128MiB_100/experiment.yaml:8]: example usage of anchors and aliases -- see `lint_steps` https://github.com/DataDog/jmxfetch/blob/master/.circleci/config.yml#L2

[the remark this answers is not in the document]
Comment (blt) [reply test/regression/cases/dogstatsd_string_interner_128MiB_100/experiment.yaml:8]: This is an area of interest in SMP as well. I'll write up a doc of options. 

But yeah, the current situation is painful.

**Comment by blt** [comment]
Capture data: [here](https://app.datadoghq.com/dashboard/x9i-4yi-27x?refresh_mode=paused&tpl_var_run-id%5B0%5D=7317c710-7a5a-421b-8a77-7f3682291078&from_ts=1697653204658&to_ts=1697653798048&live=false)

**Review by scottopell — APPROVED** [review]

**Review by GeorgeHahn — APPROVED** [review]
Experiments & the resulting data look good

**Pull request merged**